test/sql/alter.sql

```diff
@@ -446,4 +446,56 @@ ALTER TABLE i4474 ALTER COLUMN time SET statistics 10;
 SET ROLE role_4474;
 INSERT INTO i4474 SELECT '2021-01-01';
 RESET ROLE;
-
+DROP TABLE i4474 CASCADE;
+DROP ROLE role_4474;
+
+-- verify that setting replica identity works and chunks inherit the
+-- root table's setting
+CREATE TABLE replid(time timestamptz, value int);
+SELECT create_hypertable('replid', 'time', chunk_time_interval => interval '1 day', create_default_indexes => false);
+
+-- replica identity set to default
+SELECT relreplident FROM pg_class WHERE relname = 'replid';
+
+INSERT INTO replid VALUES ('2023-01-01', 1);
+-- the new chunk should have the same replica identity setting
+SELECT relname, relreplident FROM show_chunks('replid') ch INNER JOIN pg_class c ON (ch = c.oid) ORDER BY relname;
+
+-- test change to replica identity full
+ALTER TABLE replid REPLICA IDENTITY FULL;
```

Comment:
What happen if we change the replica identity for an specific chunk?

Reply:
I know this effect but I'm wondering if we should block this operation. Now we're propagating the replica identity from parent to all child tables so make sense to don't allow change it on child tables.

Reply:
How are similar things handled until now? I'm all for blocking that on a specific chunk (since it really doesn't make sense).

Reply:
We can block it, but not sure why we need to do it. I guess it is about people not shooting themselves in the foot, but what is the risk of that? Maybe it is better to leave the flexbility to let people set this the way they wish?

Reply:
I think we can leave it as is. I don't see a reason a user would do this 👍

Reply:
Actually, altering on chunks was already blocked by default, so leaving it like that for now. Added test for that case.